trustmanager/keystore.go

```diff
@@ -319,35 +319,28 @@ func getSubdir(alias string) string {
 func GetPasswdDecryptBytes(passphraseRetriever notary.PassRetriever, pemBytes []byte, name, alias string) (data.PrivateKey, string, error) {
 	var (
 		passwd  string
-		retErr  error
 		privKey data.PrivateKey
 	)
 	for attempts := 0; ; attempts++ {
 		var (
 			giveup bool
 			err    error
 		)
+		if attempts > 10 {
+			return nil, "", ErrAttemptsExceeded{}
+		}
 		passwd, giveup, err = passphraseRetriever(name, alias, false, attempts)
 		// Check if the passphrase retriever got an error or if it is telling us to give up
 		if giveup || err != nil {
 			return nil, "", ErrPasswordInvalid{}
 		}
-		if attempts > 10 {
-			return nil, "", ErrAttemptsExceeded{}
-		}
 
 		// Try to convert PEM encoded bytes back to a PrivateKey using the passphrase
 		privKey, err = utils.ParsePEMPrivateKey(pemBytes, passwd)
-		if err != nil {
-			retErr = ErrPasswordInvalid{}
-		} else {
+		if err == nil {
 			// We managed to parse the PrivateKey. We've succeeded!
-			retErr = nil
 			break
 		}
 	}
-	if retErr != nil {
```

Comment:
since we only exit the loop to reach this part if the error is nil and if it isn't then we will end up returning with ErrAttemptsExceeded or ErrPasswordInvalid, I thought this check was unnecessary and not hitting this part. So the retErr should always be nil over here. Am I missing a case?